lib/infrastructure/device/device_presenter.dart

```diff
@@ -28,27 +27,59 @@ class DeviceBindingPresenter {
       return DeviceBindingChallengeVerifiedViewModel(
         thisDevice: deviceBindingState.thisDevice,
       );
+    } else if (deviceBindingState is DeviceBindingVerificationErrorState) {
+      return DeviceBindingVerificationErrorViewModel(
+        deviceId: deviceBindingState.deviceId,
+      );
+    } else if (deviceBindingState is DeviceBindingNotPossibleState) {
+      return DeviceBindingNotPossibleViewModel(
+        reason: deviceBindingState.reason,
+      );
     }
+
     return DeviceBindingInitialViewModel();
   }
 }
 
 class DeviceBindingViewModel extends Equatable {
   final List<Device>? devices;
   final Device? thisDevice;
+  final String? deviceId;
+  final bool? isBoundDevice;
+  final bool? isBindingPossible;
+  final DeviceBindingNotPossibleReason? reason;
```

Comment:
I think that deviceId, isBoundDevice, and reason can be defined only on the "fetched" view models (or the view models where are used) 

For example, the `reason` property could be defined only on the error view model